### PR TITLE
[Tuning] MacOS DR Tuning PR

### PR DESCRIPTION
**Pull request by DefSecSentinel**
MacOS DR Tuning PR.

- Excludes macOS from a few cross-platform rules I don't think are useful or are extremely noisy. 
- Removes known FP patterns. 
- Makes some rules more specific to ensure they detect what they were meant to. 
- Brings some rules closer to parity with Endpoint rules.

### CODE REVIEW

File: rules/cross-platform/persistence_ssh_authorized_keys_modification.toml [hunk 1]
```diff
@@ -32,6 +32,10 @@ event.category:file and event.type:(change or creation) and
               /usr/local/Cellar/maven/*/libexec/bin/mvn or
               /Library/Java/JavaVirtualMachines/jdk*.jdk/Contents/Home/bin/java or
               /usr/bin/vim or
+              /opt/jc/bin/jumpcloud-agent or
+              /Library/Addigy/ssh or
+              /Applications/Parallels Desktop.app/Contents/MacOS/Parallels VM.app/Contents/MacOS/prl_vm_app or
```
Comment (Aegrah) [inline rules/cross-platform/persistence_ssh_authorized_keys_modification.toml:37]:
```suggestion
              "/Applications/Parallels Desktop.app/Contents/MacOS/Parallels VM.app/Contents/MacOS/prl_vm_app" or
```

File: rules/macos/credential_access_mitm_localhost_webproxy.toml [hunk 2]
```diff
@@ -34,7 +34,10 @@ event.category:process and host.os.type:macos and event.type:start and
  process.name : networksetup and process.args : (("-setwebproxy" or "-setsecurewebproxy" or "-setautoproxyurl") and not (Bluetooth or off)) and
  not process.parent.executable : ("/Library/PrivilegedHelperTools/com.80pct.FreedomHelper" or
                                   "/Applications/Fiddler Everywhere.app/Contents/Resources/app/out/WebServer/Fiddler.WebUi" or
-                                  "/usr/libexec/xpcproxy")
+                                  "/usr/libexec/xpcproxy") and
+ not process.Ext.effective_parent.executable : ("/Applications/GreenVPN.app/Contents/MacOS/GreenVPN", 
+                                                "/Applications/Proxyman.app/Contents/MacOS/Proxyman",
+                                                "/private/var/folders/*/AppTranslocation/*/Fiddler Everywhere.app/Contents/MacOS/Fiddler Everywhere")
```
Comment (Aegrah) [inline rules/macos/credential_access_mitm_localhost_webproxy.toml:40]: This one is not going to work, as its a KQL query, wildcards will not work when put between quotes.

Comment (Aegrah) [inline rules/macos/credential_access_mitm_localhost_webproxy.toml:40]:
```suggestion
 not process.Ext.effective_parent.executable : ("/Applications/GreenVPN.app/Contents/MacOS/GreenVPN" or 
                                                "/Applications/Proxyman.app/Contents/MacOS/Proxyman" or
                                                "/private/var/folders/*/AppTranslocation/*/Fiddler Everywhere.app/Contents/MacOS/Fiddler Everywhere")
```

File: rules/macos/credential_access_promt_for_pwd_via_osascript.toml [hunk 3]
```diff
@@ -33,11 +33,17 @@ timestamp_override = "event.ingested"
 type = "eql"
 
 query = '''
-process where host.os.type == "macos" and event.type in ("start", "process_started") and process.name : "osascript" and
- process.command_line : "osascript*display dialog*password*"
+process where event.action == "exec" and
+ process.name : "osascript" and process.args : "-e" and process.command_line : "*osascript*display*dialog*password*" and
+ not (process.parent.executable : "/usr/bin/sudo" and process.command_line : "*Encryption Key Escrow*") and
+ not (process.command_line : "*-e with timeout of 3600 seconds*" and user.id == 0 and process.parent.executable : "/bin/bash") and
```
Comment (Aegrah) [inline rules/macos/credential_access_promt_for_pwd_via_osascript.toml:39]:
```suggestion
 not (process.command_line : "*-e with timeout of 3600 seconds*" and user.id == "0" and process.parent.executable : "/bin/bash") and
```

File: rules/macos/defense_evasion_modify_environment_launchctl.toml [hunk 4]
```diff
@@ -42,7 +42,10 @@ event.category:process and host.os.type:macos and event.type:start and
   not process.parent.executable:("/Applications/IntelliJ IDEA CE.app/Contents/jbr/Contents/Home/lib/jspawnhelper" or 
                                   /Applications/NoMachine.app/Contents/Frameworks/bin/nxserver.bin or 
                                   /Applications/NoMachine.app/Contents/Frameworks/bin/nxserver.bin or 
-                                  /usr/local/bin/kr)
+                                  /usr/local/bin/kr) and
+  not process.Ext.effective_parent.executable : ("/Library/Application Support/JAMF/Jamf.app/Contents/MacOS/JamfDaemon.app/Contents/MacOS/JamfDaemon",
+                                                 "/Library/Developer/PrivateFrameworks/CoreSimulator.framework/Versions/A/XPCServices/SimulatorTrampoline.xpc/Contents/MacOS/SimulatorTrampoline",
+                                                 "/usr/bin/sshd")
```
Comment (Aegrah) [inline rules/macos/defense_evasion_modify_environment_launchctl.toml:48]:
```suggestion
  not process.Ext.effective_parent.executable : ("/Library/Application Support/JAMF/Jamf.app/Contents/MacOS/JamfDaemon.app/Contents/MacOS/JamfDaemon" or
  "/Library/Developer/PrivateFrameworks/CoreSimulator.framework/Versions/A/XPCServices/SimulatorTrampoline.xpc/Contents/MacOS/SimulatorTrampoline" or
"/usr/bin/sshd")
```

File: rules/macos/privilege_escalation_local_user_added_to_admin.toml [hunk 5]
```diff
@@ -27,10 +27,12 @@ type = "query"
 
 query = '''
 event.category:process and host.os.type:macos and event.type:(start or process_started) and
- process.name:(dscl or dseditgroup) and process.args:(("/Groups/admin" or admin) and ("-a" or "-append"))
+ process.name:(dscl or dseditgroup) and process.args:(("/Groups/admin" or admin) and ("-a" or "-append")) and
+ not process.Ext.effective_parent.executable : ("/Library/Application Support/JAMF/Jamf.app/Contents/MacOS/JamfManagementService.app/Contents/MacOS/JamfManagementService",
+                                                "/Library/Application Support/JAMF/Jamf.app/Contents/MacOS/JamfDaemon.app/Contents/MacOS/JamfDaemon",
+                                                "/usr/local/jamf/bin/jamf")
```
Comment (Aegrah) [inline rules/macos/privilege_escalation_local_user_added_to_admin.toml:33]:
```suggestion
 not process.Ext.effective_parent.executable : ("/Library/Application Support/JAMF/Jamf.app/Contents/MacOS/JamfManagementService.app/Contents/MacOS/JamfManagementService" or
                                                "/Library/Application Support/JAMF/Jamf.app/Contents/MacOS/JamfDaemon.app/Contents/MacOS/JamfDaemon" or
                                                "/usr/local/jamf/bin/jamf")
```

**Comment by Aegrah** [comment]
Fixed some unit testing issues, but it still keeps failing on new unit testing issues. Would recommend running local unit testing prior to publishing the branch next time!

File: rules/cross-platform/defense_evasion_timestomp_touch.toml [hunk 6]
```diff
@@ -33,10 +33,11 @@ query = '''
 process where event.type == "start" and
  process.name : "touch" and user.id != "0" and
  process.args : ("-r", "-t", "-a*","-m*") and
- not process.args : ("/usr/lib/go-*/bin/go", "/usr/lib/dracut/dracut-functions.sh", "/tmp/KSInstallAction.*/m/.patch/*")
+ not process.args : ("/usr/lib/go-*/bin/go", "/usr/lib/dracut/dracut-functions.sh", "/tmp/KSInstallAction.*/m/.patch/*") and
+ not process.command_line : "touch -a ../../../../ios/build/generated/ios/./FBReactNativeSpec/FBReactNativeSpec.h ../../../../ios/build/generated/ios/./FBReactNativeSpec/FBReactNativeSpec-generated.mm" and
```
Comment (brokensound77) [inline rules/cross-platform/defense_evasion_timestomp_touch.toml:37]: This seems too specific. Anything better to exclude on, like maybe the parent calling touch?

**Pull request closed without merging**